core/ui_mixins/quick_panel.py

```diff
@@ -255,7 +255,7 @@ def _on_highlight(self, index):
     def _on_selection(self, index):
         if index == self.limit:
             self.skip = self.skip + self.limit
-            sublime.set_timeout(self.show, 10)
```

Comment:
The blocking next page would freeze the UI. Why do you prefer blocking? Implementing <kbd>esc</kbd> would be challenging.